src/Factories/Transform/UserSchema/OptionalIfTransformFactory.cs

```diff
@@ -0,0 +1,36 @@
+using System.Linq;
+using System.Threading.Tasks;
+using form_builder.Models;
+
+namespace form_builder.Factories.Transform.UserSchema
+{
+    public class OptionalIfTransformFactory : IUserPageTransformFactory
+    {
+        public async Task<Page> Transform(Page page, FormAnswers convertedAnswers)
+        {
+            var optionalIfElements = page.Elements
+                .Where(element => !string.IsNullOrEmpty(element.Properties.OptionalIfValue) 
+                || !string.IsNullOrEmpty(element.Properties.OptionalIfNotValue));
+
+            if (optionalIfElements.Any())
+            {
+                foreach (var element in optionalIfElements)
+                {
+                    var userChoice = convertedAnswers.AllAnswers
+                        .Where(answer => answer.QuestionId.Equals(element.Properties.OptionalIfQuestionId))
+                        .FirstOrDefault();
+
+                    if (userChoice is not null)
+                    {
+                        if (userChoice.Response.ToString().Equals(element.Properties.OptionalIfValue) 
+                            || !string.IsNullOrEmpty(element.Properties.OptionalIfNotValue) 
+                            && !(userChoice.Response.ToString().Equals(element.Properties.OptionalIfNotValue)))
+                            element.Properties.Optional = true;
```

Comment:
Curlys. You used them above for a simple IF statement, but this muti-line IF doesn't.  I think, first choice would be to say if it's a multi-line IF, then use the curlys.. just to help others read it easier.